src/device.rs

```diff
@@ -139,4 +140,31 @@ impl<T: UsbContext> Device<T> {
     pub fn port_number(&self) -> u8 {
         unsafe { libusb_get_port_number(self.device.as_ptr()) }
     }
+
+    ///  Get the list of all port numbers from root for the specified device
+    pub fn port_numbers(&self) -> Result<Vec<u8>, Error> {
+        // As per the USB 3.0 specs, the current maximum limit for the depth is 7.
+        let mut ports = vec![0;7];
+
+        let result = unsafe {
+            libusb_get_port_numbers(
+                self.device.as_ptr(),
+                ports.as_mut_ptr(),
+                ports.len() as i32
+            )
+        };
+
+        let ports_number = if result < 0 {
+            return Err(error::from_libusb(result));
+        } else {
+            result
+        };
+        if ports_number as usize > ports.len() {
+            return Err(error::Error::Overflow);
+        }
+
+        ports.drain(0..ports.len() - ports_number as usize);
```

Comment:
Are you sure it's that what you want?
if libusb_get_port_numbers return 4 and ports vec contain vec![1, 2, 3, 4, 0, 0, 0];
after call `ports.drain(0..ports.len() - 4 as usize);` we have ports = [  0,  0,  0,]

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=53785a81342c55d2ce0e4644d0c580d8

Reply:
Maybe `ports.drain(ports_number..);`

Reply:
You are right. I pushed the wrong fix: 
I have tested on a simple example which gave me in the original buffer:
```
[2020-10-07T16:32:42Z INFO  test_usb_parent] Device 1 22
[2020-10-07T16:32:42Z INFO  test_usb_parent] Found!
xxx [6, 3, 0, 0, 0, 6, 3]
[2020-10-07T16:32:42Z INFO  test_usb_parent] Found Ok([6, 3])
```
I took the last part of the buffer, but the libusb code clearly does the memmove to put the data at the beginning of the buffer.
Sorry for this.